crates/solvers/src/boundary/legacy.rs

```diff
@@ -0,0 +1,549 @@
+use {
+    crate::{
+        boundary,
+        domain::{auction, eth, liquidity, order, solution, Solver},
+        infra::config::legacy::LegacyConfig,
+    },
+    anyhow::{Context as _, Result},
+    ethereum_types::{H160, U256},
+    futures::future::{BoxFuture, FutureExt},
+    model::order::{OrderKind, OrderUid},
+    shared::{
+        http_solver::{
+            gas_model::GasModel,
+            model::{
+                AmmModel,
+                AmmParameters,
+                BatchAuctionModel,
+                ConcentratedPoolParameters,
+                ConstantProductPoolParameters,
+                MetadataModel,
+                OrderModel,
+                SettledBatchAuctionModel,
+                StablePoolParameters,
+                TokenAmount,
+                TokenInfoModel,
+                WeightedPoolTokenData,
+                WeightedProductPoolParameters,
+            },
+            DefaultHttpSolverApi,
+            HttpSolverApi,
+            SolverConfig,
+        },
+        sources::uniswap_v3::{
+            graph_api::Token,
+            pool_fetching::{PoolInfo, PoolState, PoolStats},
+        },
+    },
+    std::collections::BTreeMap,
+};
+
+pub struct Legacy {
+    pub solver: DefaultHttpSolverApi,
+    pub weth: eth::WethAddress,
+}
+
+impl Legacy {
+    pub fn new(config: LegacyConfig) -> Self {
+        Self {
+            solver: DefaultHttpSolverApi {
+                name: config.solver_name,
+                network_name: format!("{:?}", config.chain_id),
+                chain_id: config.chain_id.value().as_u64(),
+                base: config.base_url,
+                client: reqwest::Client::new(),
+                config: SolverConfig {
+                    // Note that we unconditionally set this to "true". This is
+                    // because the auction that we are solving for already
+                    // contains which tokens can and can't be internalized,
+                    // and we don't need to duplicate this setting here. Ergo,
+                    // in order to disable internalization, the driver would be
+                    // configured to have 0 trusted tokens.
+                    use_internal_buffers: Some(true),
+                    max_nr_exec_orders: config.max_nr_exec_orders,
```

Comment:
(regarding comment ☝️) Specifically, here I would just:
```suggestion
```

The `SolverConfig: Default` implementation already sets this to 100 AFAIU.